lib/collectors.js

```diff
@@ -2,6 +2,46 @@
 
 module.exports = (api) => {
 
+  api.metasync.collect = (expected) => {
+
+    const isCount = typeof(expected) === 'number';
+    const isKeys = Array.isArray(expected);
+    if (!(isCount || isKeys)) throw new Error('Unexpected type');
+    const keys = expected;
+    if (isKeys) expected = expected.length;
+    let count = 0;
+    let timeout = 0;
+    let timer = null;
+    let done = null;
+    const data = {};
+
+    const collector = (key, value) => {
+      if (isCount || keys.includes(key)) count++;
+      if (value instanceof Error) return done(value, data);
```

Comment:
Can we use the normal errback contract here instead of fiddling with types? It would play nicely with async operations, and, besides that, `instanceof` is slow (or, at least, used to be tremendously slow not a long time ago; not sure about the most recent V8 versions though).